modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpResourceDataElement.java

```diff
@@ -0,0 +1,233 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.net.http;
+
+import org.ballerinalang.net.uri.DispatcherUtil;
+import org.ballerinalang.net.uri.parser.DataElement;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.wso2.carbon.transport.http.netty.message.HTTPCarbonMessage;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Http Node Item for URI template tree.
+ */
+public class HttpResourceDataElement implements DataElement<HttpResource, HTTPCarbonMessage> {
+
+    private List<HttpResource> resource;
+    private boolean isFirstTraverse = true;
+
+
+    @Override
+    public void setData(HttpResource newResource) {
+        if (isFirstTraverse) {
+            this.resource = new ArrayList<>();
+            this.resource.add(newResource);
+            isFirstTraverse = false;
+            return;
+        }
+        List<String> newMethods = newResource.getMethods();
+        if (newMethods == null) {
+            for (HttpResource previousResource : this.resource) {
+                if (previousResource.getMethods() == null) {
+                    //if both resources do not have methods but same URI, then throw following error.
+                    throw new BallerinaException("Seems two resources have the same addressable URI, "
```

Comment:
Do we really need the word seems here? Can't it be "two resources have the same addressable URI,", otherwise it looks like some uncertain log info